symbolic-sourcemapcache/src/lookup.rs

```diff
@@ -0,0 +1,353 @@
+        self.line
+    }
+
+    /// The number of the source column.
+    pub fn column(&self) -> u32 {
+        self.column
+    }
+
+    /// The contents of the source line.
+    pub fn line_contents(&self) -> Option<&'data str> {
+        self.file().and_then(|file| file.line(self.line as usize))
+    }
+
+    /// The scope containing this source location.
+    pub fn scope(&self) -> ScopeLookupResult<'data> {
+        self.scope
+    }
+
+    /// The name of the source file this location belongs to.
+    pub fn file_name(&self) -> Option<&'data str> {
+        self.file.and_then(|file| file.name)
+    }
+
+    /// The source of the file this location belongs to.
+    pub fn file_source(&self) -> Option<&'data str> {
+        self.file.and_then(|file| file.source)
+    }
+}
+
+type Result<T, E = Error> = std::result::Result<T, E>;
+
+/// A cached SourceMap lookup index.
+///
+/// This allows quick lookup inside SourceMaps via the [`lookup`](Self::lookup) method.
+#[derive(Clone)]
+pub struct SourceMapCache<'data> {
+    header: &'data raw::Header,
+    min_source_positions: &'data [raw::MinifiedSourcePosition],
+    orig_source_locations: &'data [raw::OriginalSourceLocation],
+    files: &'data [raw::File],
+    line_offsets: &'data [raw::LineOffset],
+    string_bytes: &'data [u8],
+}
+
+impl<'data> std::fmt::Debug for SourceMapCache<'data> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("SourceMapCache")
+            .field("version", &self.header.version)
+            .field("mappings", &self.header.num_mappings)
+            .field("files", &self.header.num_files)
+            .field("line_offsets", &self.header.num_line_offsets)
+            .field("string_bytes", &self.header.string_bytes)
+            .finish()
+    }
+}
+
+impl<'data> SourceMapCache<'data> {
+    /// Parses a raw buffer containing a serialized [`SourceMapCache`].
+    #[tracing::instrument(level = "trace", name = "SourceMapCache::parse", skip_all)]
+    pub fn parse(buf: &'data [u8]) -> Result<Self> {
+        let (header, buf) = raw::Header::ref_from_prefix(buf).ok_or(Error::Header)?;
+
+        if header.magic == raw::SOURCEMAPCACHE_MAGIC_FLIPPED {
+            return Err(Error::WrongEndianness);
+        }
+        if header.magic != raw::SOURCEMAPCACHE_MAGIC {
+            return Err(Error::WrongFormat);
+        }
+        if header.version != raw::SOURCEMAPCACHE_VERSION {
+            return Err(Error::WrongVersion);
+        }
+
+        let (_, buf) = align_to(buf, 8).ok_or(Error::SourcePositions)?;
+        let num_mappings = header.num_mappings as usize;
+        let (min_source_positions, buf) =
+            raw::MinifiedSourcePosition::slice_from_prefix(buf, num_mappings)
+                .ok_or(Error::SourcePositions)?;
+        let (orig_source_locations, buf) =
+            raw::OriginalSourceLocation::slice_from_prefix(buf, num_mappings)
+                .ok_or(Error::SourceLocations)?;
```

Comment:
the writer does an `align(8)` in between these two tables, the reader should reflect that.